File: Java/0001_Two_Sum.java
// 2020-08-03 23:59:27

import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int numNeeded = target - nums[i];
            if (map.containsKey(numNeeded)) {
                return new int[] { map.get(numNeeded), i };
            }
            map.put(nums[i], i);
        }
        return new int[] { -1, -1 };
    }
}

File: TypeScript/0001-two-sum.ts
// 2020-08-04 17:45:35

function twoSum(nums: number[], target: number): number[] {
  if (nums.length == 0) return [-1, -1];
  const map = new Map();

  for (let i = 0; i < nums.length; i++) {
    let diff = target - nums[i];
    if (map.has(diff)) {
      return [map.get(diff), i];
    } else {
      map.set(nums[i], i);
    }
  }
  return [-1, -1];
}
